ipld/write.go

```diff
@@ -38,7 +38,7 @@ func PutData(ctx context.Context, shares [][]byte, adder ipld.NodeAdder) (*rsmt2
 // ExtractODSShares returns the original shares of the given ExtendedDataSquare. This
 // is a helper function for circumstances where PutData must be used after the EDS has already
 // been generated.
-func ExtractODSShares(eds *rsmt2d.ExtendedDataSquare) [][]byte {
+func ExtractODSShares(eds *rsmt2d.ExtendedDataSquare) []Share {
```

Comment:
This is a departure from how we've done things so far, which is just use `[]byte` to represent share. It's something that's disappointed me since we've had countless instances of confusion with what exactly a "share" is (does it contain NIDs, etc.). While I approve of this change in general, it should be done everywhere rather than only in a single case.

Reply:
Can you point me to the place where similar change is not done, so I missed it?

Reply:
Yeah I was referring to the NMT and rsmt2d, which use flat byte slices. rsmt2d has an excuse that it doesn't know about namespaces, but the NMT does and should therefore use `Share` rather than `[]byte`. Well, it's a change for another day.